### PR TITLE
Enable rulesets (BLE and RUF)

### LINKED ISSUE #2567: linting: Enable more ruff rulesets
ruff linter has 50 or so [rulesets](https://docs.astral.sh/ruff/rules/) that can be enabled. Currently we enable 6 good ones, but could enable more.

### Instructions

* Pick a ruleset that is not yet checked in the list below
* Add the ruleset in `select` array in pyproject.toml
* Run `tox -e lint`
* If there are any linter complaints:
  * If the complaints seems reasonable, try fixing the issue in code 
  * If there are only a small number of errors and they don't seem like they should be fixed, it may be appropriate to add an ignore comment to those lines (example: `# noqa: F401`)
  * If there are lots of failures, it may be an option to add a specific rule to `ignore` or `per-file-ignores` in pyproject.toml. Feel free to pick another ruleset as well: some are much trickier than others
  * If you are not sure how to proceed please file a new issue, mention which ruleset you are enabling and paste the `tox -e lint` output there to get advice
* Once there are no more linter errors, create a PR with your changes!

### Rulesets

Here's a list of rulesets that looked potentially interesting to me (checked ones are already enabled):
* [x] Pyflakes (F)
* [x] pycodestyle Error (E)
* [x] isort (I)
* [x] pep8-naming (N)
* [x] pydocstyle (D)
* [x] Pylint (PL)
* [x] pyupgrade (UP)
* [x] Ruff-specific rules (RUF)
* [x] pycodestyle Warning (W)
* [x] flake8-bandit (S)
* [x] flake8-annotations (ANN)
* [x] flake8-blind-except (BLE)
* [x] flake8-bugbear (B)
* [x] flake8-builtins (A)
* [x] flake8-datetimez (DTZ)
* [x] flake8-comprehensions (C4)
* [x] flake8-implicit-str-concat (ISC)
* [x] flake8-return (RET)
* [x] flake8-self (SLF)
* [x] flake8-simplify (SIM)
* [x] flake8-unused-arguments (ARG)

### PR DESCRIPTION
**Pull request by NicholasTanz**
…UF012

**Description of the changes being introduced by the pull request**:

Fixes #2567

### CODE REVIEW

**Comment by jku** [comment]
FYI the failure in `with-sslib-main` is expected at the moment and will not prevent merging

**Comment by NicholasTanz** [comment]
> FYI the failure in `with-sslib-main` is expected at the moment and will not prevent merging

ah gotcha, can you take a look at this pr then, when you get a chance @jku.

File: pyproject.toml [hunk 1]
```diff
@@ -107,9 +109,10 @@ ignore = [
 
 [tool.ruff.lint.per-file-ignores]
 "tests/*" = [
-    "D",    # pydocstyle: no docstrings required for tests
-    "E501", # line-too-long: embedded test data in "fmt: off" blocks is ok
-    "S",    # bandit: Not running bandit on tests
+    "D",      # pydocstyle: no docstrings required for tests
+    "E501",   # line-too-long: embedded test data in "fmt: off" blocks is ok
+    "S",      # bandit: Not running bandit on tests
+    "RUF012", # ruff: mutable-class-default
```
Comment (jku) [inline pyproject.toml:115]: I guess we have a bunch of these? Could mention that in the comment but no need to modify just for that

**Review by jku — APPROVED** [review]
Looks good -- or rather the amount of `BLE001` looks absolutely awful but I think I can't blame you for that...

Thanks!

**Comment by jku** [comment]
There's "15 expected checks" listed but that's because the check names changed  in #2610 ...

All of the relevant tests have passed so I'll merge bypassing branch protections.

**Pull request merged**